lib/em/protocols/linetext2.rb

```diff
@@ -56,15 +56,22 @@ def receive_data data
 
         while remaining_data.length > 0
           if @lt2_mode == :lines
-            if ix = remaining_data.index( @lt2_delimiter )
+            delimiter_string = case @lt2_delimiter 
+            when Regexp
+              remaining_data.slice(@lt2_delimiter)
+            else
+              @lt2_delimiter
+            end
+            ix = (delimiter_string.nil?) ? nil : remaining_data.index(delimiter_string)
```

Comment:
This might be better as `ix = remaining_data.index(delimiter_string) if delimiter_string`